easy_phi/app.py

```diff
@@ -213,18 +236,24 @@ def get(self):
         widgets = scpi2widgets.scpi2widgets(supported_commands)
 
         self.set_header('Content-type', 'application/javascript')
+
+        temp_id = int(time.time()*1000)
```

Comment:
what happends in this method:
- we get array of javascript pieces (widgets). We want to execute them in isolated environment, so they will not declare global variables, mess with other widgets and so on. Also, if there is syntax error, it should not fail the whole script.
- also, every widget should have access only to its placeholder.
So, we produce a new placeholder, pass it to widget and wrap call into anonymous function (i.e. create new scope) and eval (to isolate failure). Result looks like that:

    // create widget placeholder
    $('#module_container').append("\<section id='some_unique_id'\>\</section\>);

     // some_unique_id is generated from timestamp in milliseconds 
    // .. and incremented for every succeeding widget
    (function(slot, container, scpi) { 
        // note that since it is wrapped into anonymous function call it is different 
        // from slot, contatiner and scpi from global scope.
        // now, since it is wrapped into eval() it will not crash the whole widget script file
        eval({widget})
     })(slot_id, $("#some_unique_id'), modified_ep_scpi)

I'll add this to comments with next commit